sdks/python/src/opik/integrations/harbor/opik_tracker.py

```diff
@@ -0,0 +1,473 @@
+) -> Optional["Job"]:
+    """Enable Opik tracking for Harbor.
+
+    Can be called two ways:
+    - track_harbor() - enables global tracking (for CLI usage)
+    - track_harbor(job) - wraps a job and enables tracking (for SDK usage)
+
+    Args:
+        job: Optional Harbor Job instance. If provided, returns the same job.
+        project_name: Opik project name. If None, uses OPIK_PROJECT_NAME env var.
+
+    Returns:
+        The job instance if provided, None otherwise.
+
+    Example:
+        >>> from opik.integrations.harbor import track_harbor
+        >>> job = Job(config)
+        >>> tracked_job = track_harbor(job)
+        >>> result = await tracked_job.run()
+    """
+    _enable_harbor_tracking(project_name=project_name)
+    return job
+
+
+def _wrap_trial_run(original: Callable, project_name: Optional[str]) -> Callable:
+    """Wrap Trial.run with tracing, feedback scores, and experiment linking."""
+
+    @track(
+        name="trial_run",
+        tags=["harbor"],
+        project_name=project_name,
+        capture_output=False,
+    )
+    @functools.wraps(original)
+    async def wrapped(self: "Trial") -> "TrialResult":
+        # Set nice trace name
+        config = self.config
+        trace_name = f"{config.agent.name}/{config.trial_name}"
+        opik_context.update_current_trace(
+            name=trace_name,
+            input={
+                "trial_name": config.trial_name,
+                "task": {
+                    "name": config.task.name
+                    if hasattr(config.task, "name")
+                    else str(config.task.path),
+                    "source": getattr(config.task, "source", None),
+                },
+                "agent": {
+                    "name": config.agent.name,
+                    "model": getattr(config.agent, "model_name", None),
+                },
+            },
+            metadata={"created_from": "harbor"},
+            tags=["harbor", config.agent.name],
+        )
+
+        # Lazily setup experiment service if not already done
+        # This ensures experiment tracking works for both SDK and CLI modes
+        if experiment_service.get_service() is None:
+            try:
+                # Use job_id for consistent experiment naming
+                experiment_name = (
+                    f"harbor-job-{str(config.job_id)[:8]}" if config.job_id else None
+                )
+                source = getattr(config.task, "source", None)
+                LOGGER.debug(
+                    "Lazily setting up experiment service: experiment_name=%s, source=%s",
+                    experiment_name,
+                    source,
+                )
+                experiment_service.setup_lazy(
+                    experiment_name=experiment_name,
+                    source=source,
+                )
+            except Exception as e:
+                LOGGER.debug("Failed to lazily setup experiment service: %s", e)
+
+        result = await original(self)
```

Comment:
There is no known type for the result which makes understanding and maintaining the following code quite hard.
It would be nice to start using `TypeVar`s or at least provide the exact types for `result` variables in the `_wrap` functions, so that syntax highlighting and linters can work with it.